en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,109 @@
+parameters when we call a block.
+
+This easy solution is not optimal a main reason, when working with `ESI`_, you will cache each block
+for each different url.
+
+``/block/1`` will be in one cache entry.
+``/block/1?foo=bar`` will be in another cache entry.
+
+Open Orchestra solution
+-----------------------
+
+At Open Orchestra, we prefer to give you the opportunity to pass only the parameters you want
+to request a block, this way, we would limit the cache entries for a single block.
+
+Backoffice work
+~~~~~~~~~~~~~~~
+
+In the Backoffice, we provide a way to store the parameter linked to the block in the block reference
+in the areas.
+
+You need to declare a service with the tag ``open_orchestra_backoffice.block_parameter.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_backoffice.block_parameter.strategy }
+
+The class should also implement the interface ``OpenOrchestra\Backoffice\BlockParameter\BlockParameterInterface``
+
+Frontoffice work
+~~~~~~~~~~~~~~~~
+
+In the Frontoffice, the request is created while you don't have any idea of the block type. You will have to create
+the request based only on the data stored in the ``blockParameter`` entry from the block reference.
+
+To analyse the parameters and complete the request, you need to declare a service with the tag ``open_orchestra_front.sub_query.strategy``
+
+.. code-block:: yaml
+
+    tags:
+        - { name: open_orchestra_front.sub_query.strategy }
+
+The class should also implement the interface ``OpenOrchestra\FrontBundle\SubQuery\SubQueryGeneratorInterface``
+
+Example
+-------
+
+Subquery strategies available
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
++------------------------------+------------------------------------------------------------+
+| Name                         | Action                                                     |
++==============================+============================================================+
+| ``DeviceSubQueryStrategy``   | Pass the header parameter ``x-ua-device``                  |
++------------------------------+------------------------------------------------------------+
+| ``PostDataSubQueryStrategy`` | Pass all data from the ``POST`` part                       |
++------------------------------+------------------------------------------------------------+
+| ``RequestSubQueryStrategy``  | Pass the specified request parameter (aliasId for instance |
++------------------------------+------------------------------------------------------------+
+
+New Subquery Strategy
+~~~~~~~~~~~~~~~~~~~~~
+
+Let's say that you want to translate the foo parameter from the request to give it to the block.
+
+You need to implement the three methods of the interface :
+
+ * ``getName`` : give the name of the strategy
+ * ``support`` : take the block parameter to decide if the strategy should be used
+ * ``generate`` : will generate the value
+
+In our case, we will check if the block parameter name is ``foo``.
```

Comment:
So a strategy is linked to a block, not to a parameter type ? The previous section introduced DeviceSubQueryStrategy, PostDataSubQueryStrategy and RequestSubQueryStrategy, whose names are not related to a block but a parameter type, it's confusing.